File: datavar.f90
program DataTypes
    implicit none

    ! Integer
    integer :: myInt
    ! Real (Float)
    real :: myFloat
    ! Double Precision
    double precision :: myDouble
    ! Character
    character(len=1) :: myChar
    ! String
    character(len=20) :: myString
    ! Logical (Boolean)
    logical :: myBoolean
    ! Array
    integer, dimension(5) :: myArray
    ! Derived Type (Struct equivalent)
    type Person
        character(len=20) :: name
        integer :: age
    end type Person
    type(Person) :: myPerson
    ! Nil (NULL pointer)
    integer, pointer :: myNullPtr

    ! Initialize variables
    myInt = 42
    myFloat = 3.14
    myDouble = 3.14159
    myChar = 'A'
    myString = "Hello, Fortran!"
    myBoolean = .true.
    myArray = (/1, 2, 3, 4, 5/)
    myPerson = Person("Alice", 25)
    myNullPtr => null()

    ! Print variables
    print *, "Integer:", myInt
    print *, "Float:", myFloat
    print *, "Double:", myDouble
    print *, "Char:", myChar
    print *, "String:", trim(myString)
    print *, "Boolean:", myBoolean
    print *, "Array:", myArray
    print *, "Struct: Name =", trim(myPerson%name), "Age =", myPerson%age
    print *, "Nil: ", associated(myNullPtr)

end program DataTypes

File: datavar.pas
program DataTypes;

type
  Person = record
    name: string[20];
    age: integer;
  end;

var
  myInt: integer;
  myFloat: real;
  myDouble: double;
  myChar: char;
  myString: string[20];
  myBoolean: boolean;
  myArray: array[1..5] of integer;
  myPerson: Person;
  myNil: Pointer;

begin
  { Initialize variables }
  myInt := 42;
  myFloat := 3.14;
  myDouble := 3.14159;
  myChar := 'A';
  myString := 'Hello, Pascal!';
  myBoolean := True;
  myArray[1] := 1; myArray[2] := 2; myArray[3] := 3; myArray[4] := 4; myArray[5] := 5;
  myPerson.name := 'Alice';
  myPerson.age := 25;
  myNil := nil;

  { Print variables }
  writeln('Integer: ', myInt);
  writeln('Float: ', myFloat:0:2);
  writeln('Double: ', myDouble:0:5);
  writeln('Char: ', myChar);
  writeln('String: ', myString);
  writeln('Boolean: ', myBoolean);
  write('Array: ');
  for var i := 1 to 5 do
    write(myArray[i], ' ');
  writeln;
  writeln('Struct: Name = ', myPerson.name, ', Age = ', myPerson.age);
  writeln('Nil: ', myNil);
end.
